src/porepy/geometry/intersections.py

```diff
@@ -1361,20 +1361,36 @@ def vector_pointset_point(a, b, tol=1e-8):
 
 def segments_polygon(
     start: np.ndarray, end: np.ndarray, poly: np.ndarray, tol: float = 1e-5
-) -> Tuple[np.ndarray, np.ndarray]:
+) -> tuple[np.ndarray, np.ndarray]:
     """Compute the internal intersection from line segments to a polygon in 3d.
     Intersections with the boundary of the polygon are not computed.
 
     Parameters:
-        start (np.ndarray, nd x num_segments): One endpoint of segments
-        end (np.ndarray, nd x num_segments): Other endpoint of segments
-        poly (np.ndarray, nd x n_vert): Vertexes of polygon.
-        tol (optional): Tolerance for the geometric computations.
+        start (shape=(nd, num_segments)): One endpoint of segments.
+        end (shape=(nd, num_segments)): Other endpoint of segments.
+        poly (shape=(nd, num_vertices)): Vertices of polygon.
+        tol (float): Tolerance for the geometric computations.
+
+    .. note: It is required that all points lie in a plane. A sanity check will
+        be performed.
 
     Returns:
-        np.ndarray, bool: Identify if a segment has intersection with the polygon
-            useful to filter the second return parameter.
-        np.ndarray, nd x num_segments: Intersection point.
+        A tuple containing a boolean array, identifying whether a segment has an
```

Comment:
Change the return values to appear separately in a list. Also I'm not quite sure about the numbered lists, in one usermeeting (I think that was the one you missed) we discussed this a bit differently, Veljko added an example regarding this to the tutorial (https://pmgbergen.github.io/porepy/html/docsrc/howto/howto-docstring.html#example_docstrings.example_function_4).

Reply:
Alright, very good to know! I misunderstood this based on the first introductory lecture. Thought it would be strictly forbidden to redundantly repeat the return type after it had been already introduced in the signature. I am welcoming this decision though very much!